packages/theming/src/utils/getColor.ts

```diff
@@ -37,57 +38,53 @@ const adjust = (color: string, expected: number, actual: number) => {
  * @param {number} [shade=DEFAULT_SHADE] A hue shade.
  * @param {Object} theme Context `theme` object.
  * @param {Number} [transparency] An alpha-channel value between 0 and 1.
- *
- * @component
  */
-export default function getColor(
-  hue: Hue,
-  shade: number = DEFAULT_SHADE,
-  theme?: DefaultTheme,
-  transparency?: number
-) {
-  let retVal;
-
-  if (isNaN(shade)) {
-    return undefined;
-  }
-
-  const palette = theme && theme.palette ? theme.palette : DEFAULT_THEME.palette;
-  const colors = theme && theme.colors ? theme.colors : DEFAULT_THEME.colors;
+export const getColor = memoize(
```

Comment:
Were there any performance concerns that prompted the `lodash.memoize` utility usage?

Reply:
Yes, calling out to these utilities results in repeat `getColor` calls in a lot of existing cases. But I think there's a massive savings potential overall.